metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/MonitoredScheduledExecutorService.java

```diff
@@ -0,0 +1,171 @@
+package com.ringcentral.platform.metrics.instrument.executors;
+
+import com.ringcentral.platform.metrics.MetricKey;
+import com.ringcentral.platform.metrics.MetricRegistry;
+import com.ringcentral.platform.metrics.histogram.Histogram;
+import com.ringcentral.platform.metrics.labels.LabelValues;
+import com.ringcentral.platform.metrics.names.MetricName;
+import com.ringcentral.platform.metrics.rate.Rate;
+import com.ringcentral.platform.metrics.timer.Stopwatch;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.*;
+import java.util.function.Function;
+
+import static com.ringcentral.platform.metrics.names.MetricName.name;
+
+/**
+ * A {@link ScheduledExecutorService} wrapper that provides the following metrics for scheduled tasks:
+ * <ul>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for tasks scheduled to run once, with the name: {@code metricKeyProvider.apply(name("scheduled", "once"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for tasks scheduled to run repetitively: {@code metricKeyProvider.apply(name("scheduled", "repetitively"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for tasks that overrun their scheduled period: {@code metricKeyProvider.apply(name("scheduled", "overrun"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.histogram.Histogram} for the percentage of the scheduled period used by task execution:
+ *       {@code metricKeyProvider.apply(name("scheduled", "period", "percent"))}</li>
+ * </ul>
+ *
+ * This class also provides the core metrics available in {@link MonitoredExecutorService}:
+ * <ul>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for submitted tasks, with the name {@code metricKeyProvider.apply(name("submitted"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.counter.Counter} for running tasks: {@code metricKeyProvider.apply(name("running"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for completed tasks: {@code metricKeyProvider.apply(name("completed"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.timer.Timer} for task idleness, measuring the time between task submission and the start of execution:
+ *       {@code metricKeyProvider.apply(name("idle"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.timer.Timer} for task execution time: {@code metricKeyProvider.apply(name("execution"))}</li>
+ * </ul>
+ *
+ * If the underlying (parent) {@link ScheduledExecutorService} is a {@link ForkJoinPool}, it additionally provides the following {@link com.ringcentral.platform.metrics.var.longVar.LongVar} metrics:
```

Comment:
It seems that it doesn't work this way - didn't find a place where those metrics are added for `MonitoredScheduledExecutorService`.
It's checked and added in `MonitoredExecutorService`, but `MonitoredScheduledExecutorService` doesn't inherit it.
Could you, please, double check it?

Reply:
You are right, it's a copy-paste. Will remove it, thanks.